README.md
# control-centre-antix

debian/changelog
control-centre-antix (0.6.7) unstable; urgency=medium

  * Updated ca, de, es, fr, nl, pt

 -- anticapitalista <antix@operamail.com>  Fri, 16 Mar 2018 17:44:48 +0200

control-centre-antix (0.6.6) unstable; urgency=medium

  * Amharic language added
  * Added Repo-manager to Maintenance

 -- anticapitalista <antix@operamail.com>  Mon, 26 Feb 2018 20:46:23 +0200

control-centre-antix (0.6.5) unstable; urgency=medium

  * jwm settings button to match icons with icewm and fluxbox
  * updated translations

 -- anticapitalista <antix@operamail.com>  Sat, 18 Nov 2017 17:02:03 +0200

control-centre-antix (0.6.4) unstable; urgency=medium

  *  installer button should not appear on installed systems

 -- anticapitalista <antix@operamail.com>  Thu, 02 Nov 2017 19:37:01 +0200

control-centre-antix (0.6.3) unstable; urgency=medium

  * Use rc-conf-wrapper.sh script - thanks skidoo

 -- anticapitalista <antix@operamail.com>  Sun, 08 Oct 2017 18:57:23 +0300

control-centre-antix (0.6.2) unstable; urgency=medium

  * Include in Applications menu

 -- anticapitalista <antix@operamail.com>  Sat, 07 Oct 2017 15:11:50 +0300

control-centre-antix (0.6.1) unstable; urgency=medium

  * Added 1to1 apps and ssh-conduit to show in control centre if installed

 -- anticapitalista <antix@operamail.com>  Sat, 16 Sep 2017 12:39:08 +0100

control-centre-antix (0.6.0) unstable; urgency=medium

  * Restructured control centre

 -- anticapitalista <antix@operamail.com>  Sat, 15 Jul 2017 12:16:59 +0100

control-centre-antix (0.5.12) unstable; urgency=medium

  * Fixed typo that disabled sound card
  * Added antiX-cli-cc

 -- anticapitalista <antix@operamail.com>  Mon, 03 Jul 2017 01:26:09 +0300

control-centre-antix (0.5.11) unstable; urgency=medium

  * Added packageinstaller and packageinstaller-pkglist to control depends

 -- anticapitalista <antix@operamail.com>  Sun, 02 Jul 2017 13:26:08 +0300

control-centre-antix (0.5.10) unstable; urgency=medium

  * Replaced install-meta package with packageinstaller

 -- anticapitalista <antix@operamail.com>  Thu, 29 Jun 2017 14:38:53 +0300

control-centre-antix (0.5.9) unstable; urgency=medium

  * removed flash-antix dependency and code

 -- anticapitalista <antix@operamail.com>  Sun, 25 Jun 2017 14:47:48 +0300

control-centre-antix (0.5.8) unstable; urgency=medium

  * Added lxterminal, artwork17-antix, arc-theme, grubcc-antix to depends for jessie-stretch upgrade 

 -- anticapitalista <antix@operamail.com>  Wed, 21 Jun 2017 18:25:05 +0300

control-centre-antix (0.5.7) unstable; urgency=medium

  * Mountbox icon changed

 -- anticapitalista <antix@operamail.com>  Thu, 27 Apr 2017 00:45:43 +0300

control-centre-antix (0.5.6) unstable; urgency=medium

  * Added User Manager

 -- anticapitalista <antix@operamail.com>  Sat, 22 Apr 2017 18:31:34 +0300

control-centre-antix (0.5.5) unstable; urgency=medium

  * Added antixccgrub.sh to control centre

 -- anticapitalista <antix@operamail.com>  Mon, 10 Apr 2017 17:12:33 +0300

control-centre-antix (0.5.4) unstable; urgency=medium

  * Updated translations

 -- anticapitalista <antix@operamail.com>  Tue, 04 Apr 2017 22:09:00 +0300

control-centre-antix (0.5.3) unstable; urgency=medium

  * Rearranged tabs to avoid gtkdiaog failures if no apps in a vbox are not installed

 -- anticapitalista <antix@operamail.com>  Sat, 25 Mar 2017 18:39:05 +0200

control-centre-antix (0.5.2) unstable; urgency=medium

  * Fix typo error for installation of ati/nvidia drivers

 -- anticapitalista <antix@operamail.com>  Tue, 21 Mar 2017 23:33:36 +0200

control-centre-antix (0.5.1) unstable; urgency=medium

  * Reduced icon set
  * antixcc.sh only show installed apps in control centre

 -- anticapitalista <antix@operamail.com>  Tue, 14 Mar 2017 15:24:02 +0200

control-centre-antix (0.5.0) unstable; urgency=medium

  * live-usb-maker-gui replaces antix2usb
  * lxkeymap to set session keyboard layout

 -- anticapitalista <antix@operamail.com>  Fri, 10 Mar 2017 19:29:27 +0200

control-centre-antix (0.4.23) unstable; urgency=medium

  * New .desktop names for icewm and jwm in stretch/sid

 -- anticapitalista <antix@operamail.com>  Wed, 08 Mar 2017 18:53:13 +0200

control-centre-antix (0.4.22) unstable; urgency=medium

  * Updated translations

 -- anticapitalista <antix@operamail.com>  Sat, 07 Jan 2017 14:37:06 +0200

control-centre-antix (0.4.21) unstable; urgency=medium

  * Added missing & to lines 65 and 503 

 -- anticapitalista <antix@operamail.com>  Thu, 30 Jun 2016 19:06:13 +0300

control-centre-antix (0.4.20) unstable; urgency=medium

  * alsamixer-equalizer appears only if present

 -- anticapitalista <antix@operamail.com>  Wed, 22 Jun 2016 00:04:33 +0300

control-centre-antix (0.4.19) unreleased; urgency=medium

  * Added Catalan 

 -- anticapitalista <antix@operamail.com>  Mon, 06 Jun 2016 11:46:06 +0300

control-centre-antix (0.4.18) unstable; urgency=medium

  * Added automount option 
  * Added set-dpi
  * Updated translations

 -- anticapitalista <antix@operamail.com>  Thu, 02 Jun 2016 09:57:15 +0300

control-centre-antix (0.4.17) unstable; urgency=medium

  * updated translations 

 -- anticapitalista <antix@operamail.com>  Mon, 30 May 2016 16:05:12 +0300

control-centre-antix (0.4.16) unstable; urgency=medium

  * fixed Data and Time bug 

 -- anticapitalista <antix@operamail.com>  Sun, 29 May 2016 20:04:42 +0300

control-centre-antix (0.4.15) unstable; urgency=medium

  * added back the names of antiX2usb and UNetbootin

antixcc.sh
#!/bin/bash
# File Name: controlcenter.sh
# Purpose: all-in-one control centre for antiX
# Authors: OU812 and minor modifications by anticapitalista
# Latest Change:
# 20 August 2008
# 11 January 2009 and renamed antixcc.sh
# 15 August 2009 some apps and labels altered.
# 09 March 2012 by anticapitalista. Added Live section.
# 22 March 2012 by anticapitalista. Added jwm config options and edited admin options.
# 18 April 2012 by anticapitalista. mountbox-antix opens as user not root.
# 06 October 2012 by anticapitalista. Function for ICONS. New icon theme.
# 26 October 2012 by anticapitalista. Includes gksudo and ktsuss.
# 12 May 2013 by anticapitalista. Let user set default apps.
# 05 March 2015 by BitJam: Add alsa-set-card, edit excludes, edit bootloader.  Fix indentation.
#   * Hide live tab on non-live systems.  Use echo instead of gettext.
#   * Remove unneeded doublequotes between tags.  Use $(...) instead of `...`.
# 01 May 2016 by anticapitalista: Use 1 script and use hides if nor present on antiX-base
# 11 July 2017 by BitJam:
#   * use a subroutine to greatly consolidate code
#   * use existence of executable as the key instead of documentation directory
#     perhaps I should switch to "which" or "type"
#   * move set-dpi to desktop tab
#   * enable ati driver button in hardware tab
# 18 Nov by antiX-Dave: fix edit jwm settings button to match icons with icewm and fluxbox
#
# Acknowledgements: Original script by KDulcimer of TinyMe. http://tinyme.mypclinuxos.com
#################################################################################################################################################

TEXTDOMAINDIR=/usr/share/locale
TEXTDOMAIN=antixcc.sh
# Options
ICONS=/usr/share/icons/antiX
ICONS2=/usr/share/pixmaps

EXCLUDES_DIR=/usr/local/share/excludes 

EDITOR="geany -i"


its_alive() {
    # return 0
    local root_fstype=$(df -PT / | tail -n1 | awk '{print $2}')
    case $root_fstype in
        aufs|overlay) return 0 ;;
                   *) return 1 ;;
    esac
}

its_alive && ITS_ALIVE=true

Desktop=$"Desktop" System=$"System" Network=$"Network" Shares=$"Shares" Session=$"Session"
Live=$"Live" Disks=$"Disks" Hardware=$"Hardware" Drivers=$"Drivers" Maintenance=$"Maintenance"
dpi_label=$(printf "%s (DPI)" $"Set Font Size")

vbox() {
    local text="$*"
    local len=${#text}
    #printf "vlen: %6s\n" "$len" >&2
    [ $len -lt 20 ] && return
    echo "  <vbox>"
    local item
    for item; do
        echo "$item"
    done
    echo "  </vbox>"
}

hbox() {
    local text="$*"
    local len=${#text}
    #printf "hlen: %6s\n" "$len" >&2
    [ $len -lt 20 ] && return
    echo "<hbox>"
    local item
    for item; do
        echo "$item"
    done
    echo "</hbox>"
}

vbox_frame_hbox() {
    local text="$*"
    local len=${#text}
    #printf "flen: %6s\n" "$len" >&2
    [ $len -lt 20 ] && return
    echo "<vbox><frame><hbox>"
    local item
    for item; do
        echo "$item"
    done
    echo "</hbox></frame></vbox>"
}

entry() {
    local image=$1  action=$2  text=$3
    cat<<Entry
    <hbox>
      <button>
        <input file>$image</input>
        <action>$action</action>
      </button>
      <text use-markup="true" width-chars="32">
        <label>$text</label>
      </text>
    </hbox>
Entry
}

[ -d $HOME/.fluxbox -a -e /usr/share/xsessions/fluxbox.desktop ] && fluxbox_entry=$(entry \
    "$ICONS/gnome-documents.png" \
    "$EDITOR $HOME/.fluxbox/overlay $HOME/.fluxbox/keys $HOME/.fluxbox/init $HOME/.fluxbox/startup $HOME/.fluxbox/apps $HOME/.fluxbox/menu &" \
    $"Edit Fluxbox Settings")

[ -d $HOME/.icewm -a -e /usr/share/xsessions/icewm-session.desktop ] && icewm_entry=$(entry \
    $ICONS/gnome-documents.png \
    "$EDITOR $HOME/.icewm/winoptions $HOME/.icewm/preferences $HOME/.icewm/keys $HOME/.icewm/startup $HOME/.icewm/toolbar $HOME/.icewm/menu &" \
    $"Edit IceWM Settings")

[ -d $HOME/.jwm -a -e /usr/share/xsessions/jwm.desktop ] && jwm_entry=$(entry \
    $ICONS/gnome-documents.png \
    "$EDITOR $HOME/.jwm/preferences $HOME/.jwm/keys $HOME/.jwm/tray $HOME/.jwm/startup $HOME/.jwmrc $HOME/.jwm/menu &" \
    $"Edit jwm Settings")

# Edit syslinux.cfg if the device it is on is mounted read-write
grep -q " /live/boot-dev .*\<rw\>" /proc/mounts && bootloader_entry=$(entry \
    $ICONS/preferences-desktop.png \
    "gksu '$EDITOR /live/boot-dev/boot/syslinux/syslinux.cfg /live/boot-dev/boot/grub/grub.cfg' &" \
    $"Edit Bootloader menu")

test -d /usr/local/share/excludes && excludes_entry=$(entry \
    $ICONS/remastersys.png \
    "gksu $EDITOR $EXCLUDES_DIR/*.list &" \
    $"Edit Exclude files")

if test -x /usr/sbin/synaptic; then synaptic_entry=$(entry \
    $ICONS2/synaptic.png \
    "gksu synaptic &" \
    $"Manage Packages")

elif test -x /usr/local/bin/cli-aptiX; then synaptic_entry=$(entry \
    $ICONS2/synaptic.png \
    "desktop-defaults-run -t sudo /usr/local/bin/cli-aptiX --pause &" \
    $"Manage Packages")
fi

test -x  /usr/sbin/bootrepair && bootrepair_entry=$(entry \
    $ICONS/computer.png \
    "gksu bootrepair &" \
    $"Boot Repair")

test -x /usr/bin/wicd-gtk && wicd_entry=$(entry \
    $ICONS/nm-device-wireless.png \
    "wicd-gtk &" \
    $"Connect Wirelessly (wicd)")

firewall_prog=/usr/bin/gufw
test -x $firewall_prog  && firewall_entry=$(entry \
    $ICONS/firewall.png \
    "gksu gufw &" \
    $"Manage Firewall")

backup_prog=/usr/bin/luckybackup
test -x $backup_prog  && backup_entry=$(entry \
    $ICONS/luckybackup.png \
    "gksu luckybackup &" \
    $"Backup Your System")

equalizer_prog=/usr/bin/alsamixer
test -x $equalizer_prog  && equalizer_entry=$(entry \
    $ICONS2/alsamixer-equalizer.png \
    "desktop-defaults-run -t alsamixer -D equal &" \
    $"Alsamixer Equalizer")

unetbootin_prog=/usr/bin/unetbootin
test -x $unetbootin_prog  && unetbootin_entry=$(entry \
    $ICONS/usb-creator.png \
    "gksu unetbootin &" \
    $"Create Live-USB retain partitions (UNetbootin)")

printer_prog=/usr/bin/system-config-printer
test -x $printer_prog  && printer_entry=$(entry \
    $ICONS2/hplj1020_icon.png \
    "system-config-printer &" \
    $"Setup a Printer")

livekernel_prog=/usr/local/bin/live-kernel-updater
test -x $livekernel_prog && livekernel_entry=$(entry \
    $ICONS/usb-creator.png \
    "desktop-defaults-run -t sudo /usr/local/bin/live-kernel-updater --pause &" \
    $"Live-usb kernel updater")

lxkeymap_prog=/usr/bin/lxkeymap
test -x $lxkeymap_prog && lxkeymap_entry=$(entry \
    $ICONS/keyboard.png \
    "lxkeymap &" \
    $"Change Keyboard Layout for Session")

fskbsetting_prog=/usr/bin/fskbsetting
test -x $fskbsetting_prog && fskbsetting_entry=$(entry \
    $ICONS/usb-creator.png \
    "gksu fskbsetting &" \
    $"Set System Keymap")

wallpaper_prog=/usr/local/bin/wallpaper.py
test -x $wallpaper_prog && wallpaper_entry=$(entry \
    $ICONS/preferences-desktop-wallpaper.png \
    "wallpaper.py &" \
    $"Choose Wallpaper")

conky_prog=/usr/bin/conky
test -x $conky_prog && test -w $HOME/.conkyrc && conky_entry=$(entry \
    $ICONS/utilities-system-monitor.png \
    "desktop-defaults-run -te $HOME/.conkyrc &" \
    $"Edit System Monitor(conky)")

lxappearance_prog=/usr/bin/lxappearance
test -x $lxappearance_prog && lxappearance_entry=$(entry \
    $ICONS/preferences-desktop-theme.png \
    "lxappearance &" \
    $"Change Gtk2 and Icon Themes")

prefapps_prog=/usr/local/bin/desktop-defaults-set
test -x $prefapps_prog && prefapps_entry=$(entry \
    $ICONS/applications-system.png \
    "desktop-defaults-set &" \
    $"Preferred Applications")

packageinstaller_prog=/usr/bin/packageinstaller
test -x $packageinstaller_prog && packageinstaller_entry=$(entry \
    $ICONS/packageinstaller.png \
    "gksu packageinstaller &" \
    $"Package Installer")

svconf_prog=/usr/sbin/sysv-rc-conf
test -x $sysvconf_prog && sysvconf_entry=$(entry \
    $ICONS/gnome-settings-default-applications.png \
    "rc-conf-wrapper.sh &" \
    $"Choose Startup Services")

tzdata_dir=/usr/share/zoneinfo
tzdata_prog=/usr/sbin/dpkg-reconfigure
test -x $tzdata_prog && test -d $tzdata_dir && tzdata_entry=$(entry \
    $ICONS/time-admin.png \
    "desktop-defaults-run -t sudo dpkg-reconfigure tzdata &" \
    $"Set Date and Time")

ceni_prog=/usr/bin/ceni
test -x $ceni_prog && ceni_entry=$(entry \
    $ICONS/network-wired.png \
    "desktop-defaults-run -t sudo ceni &" \
    $"Network Interfaces (ceni)")

umts_prog=/usr/bin/umts-panel
test -x $umts_prog && umts_entry=$(entry \
    $ICONS/network-wired.png \
    "umts-panel &" \
    $"Configure GPRS/UMTS")

connectshares_prog=/usr/local/bin/connectshares-config
test -x $connectshares_prog && connectshares_entry=$(entry \
    $ICONS/connectshares-config.png \
    "connectshares-config &" \
    $"Configure Connectshares")

disconnectshares_prog=/usr/local/bin/disconnectshares
test -x $disconnectshares_prog && disconnectshares_entry=$(entry \
    $ICONS/connectshares.png \
    "disconnectshares &" \
    $" Disconnectshares")

droopy_prog=/usr/local/bin/droopy.sh
test -x $droopy_prog && droopy_entry=$(entry \
    $ICONS2/droopy.png \
    "droopy.sh &" \
    $"Share Files via Droopy")

assistant_prog=/usr/local/bin/1-to-1_assistance.sh
test -x $assistant_prog && assistant_entry=$(entry \
    $ICONS2/1-to-1_assistance.png \
    "1-to-1_assistance.sh &" \
    $"1-to-1_assistance")

voice_prog=/usr/local/bin/1-to-1_voice.sh
test -x $voice_prog && voice_entry=$(entry \
    $ICONS2/1-to-1_voice.png \
    "1-to-1_voice.sh &" \
    $"1-to-1_voice")

sshconduit_prog=/usr/local/bin/ssh-conduit.sh
test -x $sshconduit_prog && sshconduit_entry=$(entry \
    $ICONS2/ssh-conduit.png \
    "ssh-conduit.sh &" \
    $"ssh-conduit")

gnomeppp_prog=/usr/bin/gnome-ppp
test -x $gnomeppp_prog && gnomeppp_entry=$(entry \
    $ICONS/internet-telephony.png \
    "gnome-ppp &" \
    $"Configure Dial-Up")

wpasupplicant_prog=/usr/sbin/wpa_gui
test -x $wpasupplicant_prog && wpasupplicant_entry=$(entry \
    $ICONS/nm-device-wireless.png \
    "/usr/sbin/wpa_gui &" \
    $"Configure wpa_supplicant")

pppoeconf_prog=/usr/sbin/pppoeconf
test -x $pppoeconf_prog && pppoeconf_entry=$(entry \
    $ICONS/internet-telephony.png \
    "desktop-defaults-run -t /usr/sbin/pppoeconf &" \
    $"ADSL/PPPOE configuration")

adblock_prog=/usr/local/bin/block-advert.sh
test -x $adblock_prog && adblock_entry=$(entry \
    $ICONS2/advert-block.png \
    "gksu block-advert.sh &" \
    $"Adblock")

slim_cc=/usr/local/bin/antixccslim.sh
slim_prog=/usr/bin/slim
test -x $slim_prog && test -x $slim_cc && slim_entry=$(entry \
    $ICONS/preferences-desktop-wallpaper.png \
    "gksu antixccslim.sh &" \
    $"Change Slim Background")

grub_prog=/usr/local/bin/antixccgrub.sh
test -x $grub_prog && grub_entry=$(entry \
    $ICONS/screensaver.png \
    "gksu antixccgrub.sh &" \
    $"Grub Boot Image (jpg only)")

which ${EDITOR%% *} &>/dev/null && confroot_entry=$(entry \
    $ICONS/gnome-documents.png \
    "gksu $EDITOR /etc/fstab /etc/default/keyboard /etc/grub.d/* /etc/slim.conf /etc/apt/sources.list.d/*.list &" \
    $"Edit Config Files")

arandr_prog=/usr/bin/arandr
test -x $arandr_prog && arandr_entry=$(entry \
    $ICONS/video-display.png \
    "gksu arandr &" \
    $"Set Screen Resolution")

gksu_prog=/usr/bin/gksu-properties
test -x $gksu_prog && gksu_entry=$(entry \
    $ICONS2/gksu.png \
    "gksu-properties &" \
    $"Password Prompt(su/sudo)")

slimlogin_prog=/usr/local/bin/slim-login
test -x $slimlogin_prog && slimlogin_entry=$(entry \
    $ICONS/preferences-system-login.png \
    "gksu slim-login &" \
    $"Set auto-login")

screenblank_prog=/usr/local/bin/set-screen-blank
test -x $screenblank_prog && screenblank_entry=$(entry \
    $ICONS/screensaver.png \
    "set-screen-blank &" \
    $"Set Screen Blanking")

desktopsession_dir=/usr/share/doc/desktop-session-antix
test -d $desktopsession_dir  && desktopsession_entry=$(entry \
    $ICONS/preferences-system-session.png \
    "$EDITOR $HOME/.desktop-session/*.conf $HOME/.desktop-session/startup &" \
    $"User Desktop-Session")

automount_prog=/usr/local/bin/automount-config
test -x $automount_prog && automount_entry=$(entry \
    $ICONS/mountbox.png \
    "automount-config &" \
    $"Configure Automounting")

mountbox_prog=/usr/local/bin/mountbox
test -x $mountbox_prog && mountbox_entry=$(entry \
    $ICONS/mountbox.png \
    "mountbox &" \
    $"Mount Connected Devices")

liveusb_prog_g=/usr/bin/live-usb-maker-gui-antix
liveusb_prog=/usr/local/bin/live-usb-maker
if test -x $liveusb_prog_g; then
liveusb_entry=$(entry \
    $ICONS/usb-creator.png \
    "gksu live-usb-maker-gui-antix &" \
    $"Create Live-USB (live-usb-maker-gui)")

elif test -x $liveusb_prog; then
liveusb_entry=$(entry \
    $ICONS/usb-creator.png \
     "desktop-defaults-run sudo &live-usb-maker &" \
     $"Create Live-USB (live-usb-maker)")
fi

installer_prog=/usr/sbin/minstall
[ -x $installer_prog -a -n "$ITS_ALIVE" ] && installer_entry=$(entry \
    $ICONS2/msystem.png \
    "gksudo $installer_prog &" \
    $"Install antiX Linux")

partimage_prog=/usr/sbin/partimage
test -x $partimage_prog && partimage_entry=$(entry \
    $ICONS/drive-harddisk-system.png \
    "desktop-defaults-run -t sudo partimage &" \
    $"Image a Partition")

grsync_prog=/usr/bin/grsync
test -x $grsync_prog && grsync_entry=$(entry \
    $ICONS/grsync.png \
    "grsync &" \
    $"Synchronize Directories")

gparted_prog=/usr/sbin/gparted
test -x $gparted_prog && gparted_entry=$(entry \
    $ICONS/gparted.png \
    "gksu gparted &" \
    $"Partition a Drive")

setdpi_prog=/usr/local/bin/set-dpi
test -x $setdpi_prog && setdpi_entry=$(entry \
    $ICONS/fonts.png \
    "set-dpi &" \
    "$dpi_label")

inxi_prog=/usr/local/bin/inxi-gui
test -x $inxi_prog && inxi_entry=$(entry \
    $ICONS2/info_blue.png \
    "inxi-gui &" \
    $"PC Information")

mouse_prog=/usr/local/bin/ds-mouse
test -x $mouse_prog && mouse_entry=$(entry \
    $ICONS/input-mouse.png \
    "ds-mouse &" \
    $"Configure Mouse")

soundcard_prog=/usr/local/bin/alsa-set-default-card
test -x $soundcard_prog && soundcard_entry=$(entry \
    $ICONS2/soundcard.png \
    "alsa-set-default-card &" \
    $"Set Default Sound Card")

mixer_prog=/usr/bin/alsamixer
test -x $mixer_prog && mixer_entry=$(entry \
    $ICONS/audio-volume-high-panel.png \
    "desktop-defaults-run -t alsamixer &" \
    $"Adjust Mixer")

test -x $ddm_prog && nvdriver_entry=$(entry \
    $ICONS2/nvidia-ddm-mx.png \
    "desktop-defaults-run -t su-to-root -c '/usr/local/bin/ddm-mx -i nvidia' &" \
    $"Nvidia Driver Installer")

ndiswrapper_prog=/usr/sbin/ndisgtk
test -x $ndiswrapper_prog && ndiswrapper_entry=$(entry \
    $ICONS/computer.png \
    "gksu /usr/sbin/ndisgtk &" \
    $"MS Windows Wireless Drivers")

snapshot_prog=/usr/bin/isosnapshot
test -x $snapshot_prog && snapshot_entry=$(entry \
    $ICONS/preferences-system.png \
    "gksu isosnapshot &" \
    $"Create Snapshot(ISO)")

soundtest_prog=/usr/bin/speaker-test
test -x $soundtest_prog  && soundtest_entry=$(entry \
    $ICONS/preferences-desktop-sound.png \
    "urxvt -e speaker-test --channels 2 --test wav --nloops 3 &" \
    $"Test Sound")

menumanager_prog=/usr/local/bin/menu_manager.sh
test -x $menumanager_prog && menumanager_entry=$(entry \
    $ICONS2/menu_manager.png \
    "sudo menu_manager.sh &" \
    $"Edit menus")

usermanager_prog=/usr/local/bin/user-management
test -x $usermanager_prog && usermanager_entry=$(entry \
    $ICONS/config-users.png \
    "gksu user-management &" \
    $"User Manager")

galternatives_prog=/usr/bin/galternatives
test -x $galternatives_prog && galternatives_entry=$(entry \
    $ICONS2/galternatives.png \
    "galternatives &" \
    $"Alternatives Configurator")

codecs_prog=/usr/sbin/codecs
test -x $codecs_prog && codecs_entry=$(entry \
    $ICONS/applications-system.png \
    "su-to-root -X -c codecs &" \
    $"Install Restricted Codecs")

broadcom_prog=/usr/sbin/broadcom-manager
test -x $broadcom_prog && broadcom_entry=$(entry \
    $ICONS/palimpsest.png \
    "su-to-root -X -c broadcom-manager &" \
    $"Network Troubleshooting")

repomanager_prog=/usr/sbin/broadcom-manager
test -x $repomanager_prog && repomanager_entry=$(entry \
    $ICONS/repo-manager.png \
    "su-to-root -X -c repo-manager &" \
    $"Manage APT Repositories")

which backlight-brightness &>/dev/null && [ -n "$(ls /sys/class/backlight 2>/dev/null)" ] \
    && backlight_entry=$(entry \
    "/usr/share/icons/Adwaita/48x48/devices/video-display-symbolic.symbolic.png" \
    "desktop-defaults-run -t backlight-brightness &" \
    $"Backlight brightness")

[ -e /etc/live/config/save-persist -o -e /etc/live/config/persist-save.conf ]  && persist_save=$(entry \
    $ICONS/palimpsest.png \
    "gksu persist-save &" \
    $"Save root persistence")

[ -e /etc/live/config/remasterable -o -e /etc/live/config/remaster-live.conf ] && live_remaster=$(entry \
    $ICONS/preferences-desktop.png \
    "gksu live-remaster &" \
    $"Remaster-Customize Live")

live_tab=$(cat<<Live_Tab
$(vbox_frame_hbox \
"$(vbox \
"$(entry "$ICONS/remastersys.png" "gksu persist-config &" $"Configure live persistence")" \
"$livekernel_entry" "$bootloader_entry" "$persist_save")" \
"$(vbox \
"$(entry $ICONS/palimpsest.png "gksu persist-makefs &" $"Set up live persistence")" \
"$excludes_entry" "$live_remaster")")
Live_Tab
)

# If we are on a live system then ...
if grep -q " /live/aufs " /proc/mounts; then
    tab_labels="$Desktop|$System|$Network|$Shares|$Session|$Live|$Disks|$Hardware|$Drivers|$Maintenance"

else
    tab_labels="$Desktop|$System|$Network|$Shares|$Session|$Disks|$Hardware|$Drivers|$Maintenance"
    live_tab=
fi

export ControlCenter=$(cat<<Control_Center
<window title="antiX Control Center" icon="gnome-control-center" window-position="1">
  <vbox>
<notebook tab-pos="0" labels="$tab_labels">
$(vbox_frame_hbox \
"$(vbox "$wallpaper_entry" "$icewm_entry" "$jwm_entry" "$conky_entry")" \
"$(vbox "$setdpi_entry" "$lxappearance_entry" "$fluxbox_entry" "$prefapps_entry")" )

$(vbox_frame_hbox \
"$(vbox "$synaptic_entry" "$packageinstaller_entry" "$sysvconf_entry" "$galternatives_entry")" \
"$(vbox "$confroot_entry" "$fskbsetting_entry" "$tzdata_entry")" )

$(vbox_frame_hbox \
"$(vbox "$ceni_entry" "$umts_entry" "$wicd_entry" "$pppoeconf_entry")" \
"$(vbox "$gnomeppp_entry" "$wpasupplicant_entry" "$firewall_entry" "$adblock_entry")" )

$(vbox_frame_hbox \
"$(vbox "$connectshares_entry" "$droopy_entry" "$assistant_entry" "$voice_entry")" \
"$(vbox "$disconnectshares_entry" "$sshconduit_entry" "$samba_entry")" )

$(vbox_frame_hbox \
"$(vbox "$lxkeymap_entry" "$slim_entry" "$arandr_entry" "$grub_entry")" \
"$(vbox "$gksu_entry" "$slimlogin_entry" "$screenblank_entry" "$desktopsession_entry")")

$live_tab

$(vbox_frame_hbox \
"$(vbox "$automount_entry" "$installer_entry" "$mountbox_entry" "$unetbootin_entry")" \
"$(vbox "$liveusb_entry" "$partimage_entry" "$grsync_entry" "$gparted_entry")")

$(vbox_frame_hbox \
"$(vbox "$printer_entry" "$inxi_entry" "$mouse_entry" "$backlight_entry")" \
"$(vbox "$soundcard_entry" "$soundtest_entry" "$mixer_entry" "$equalizer_entry")")

$(vbox_frame_hbox \
"$(vbox "$nvdriver_entry" "$atidriver_entry" "$codecs_entry")" \
"$(vbox "$ndiswrapper_entry")" )

$(vbox_frame_hbox \
"$(vbox "$snapshot_entry" "$backup_entry" "$broadcom_entry" "$repomanager_entry")" \
"$(vbox "$bootrepair_entry" "$menumanager_entry" "$usermanager_entry")" )

</notebook>
</vbox>
</window>
Control_Center
)

case $1 in
    -d|--debug) echo "$ControlCenter" ; exit ;;
esac

gtkdialog --program=ControlCenter
#unset ControlCenter
